pandas/tests/groupby/test_apply.py

```diff
@@ -974,3 +974,24 @@ def test_apply_function_with_indexing_return_column():
     result = df.groupby("foo1", as_index=False).apply(lambda x: x.mean())
     expected = DataFrame({"foo1": ["one", "three", "two"], "foo2": [3.0, 4.0, 4.0]})
     tm.assert_frame_equal(result, expected)
+
+
+def test_apply_function_called_count(capsys):
```

Comment:
move this near test_group_apply_once_per_group and indicate the same issue number (as well)

Reply:
and rename these test to test_group_apply_once_per_group2

Reply:
Alright @jreback . Will change it accordingly

Reply:
Should remove `GH: 31111` and use `# GH2936, GH7739, GH10519, GH2656, GH12155, GH20084, GH21417` @jreback

Reply:
you can add this issue number as well is ok

Reply:
Alright. fixed accordingly